nupic/research/frameworks/vernon/handlers.py

```diff
@@ -1059,14 +1056,277 @@ def get_execution_order(cls):
+
+        return results
+
+    def run_task(self, task, cloned_adaptation_net):
+        self.train_fast_loader.sampler.set_active_tasks(task)
+        self.val_fast_loader.sampler.set_active_tasks(task)
+
+        # Train, one batch
+        data, target = next(iter(self.train_fast_loader))
+        data = data.to(self.device)
+        target = target.to(self.device)
+        train_loss = self._loss_function(
+            cloned_adaptation_net(data), target
+        )
+        # Update in place
+        self.adapt(cloned_adaptation_net, train_loss)
+
+        # Evaluate the adapted model
+        with torch.no_grad():
+            data, target = next(iter(self.val_fast_loader))
+            data = data.to(self.device)
+            target = target.to(self.device)
+
+            preds = cloned_adaptation_net(data)
+            valid_error = self._loss_function(preds, target)
+            valid_error /= len(data)
+            self.logger.debug(f"Valid error meta train training: {valid_error}")
+
+            # calculate accuracy
+            preds = preds.argmax(dim=1).view(target.shape)
+            valid_accuracy = (preds == target).sum().float() / target.size(0)
+            self.logger.debug(f"Valid accuracy meta train training: {valid_accuracy}")
+
+    @classmethod
+    def update_params(cls, params, loss, lr, distributed=False):
+        """
+        Takes a gradient step on the loss and updates the cloned parameters in place.
+        """
+        gradients = torch.autograd.grad(
+            loss, params,
+            retain_graph=True, create_graph=True
+        )
+
+        if distributed:
+            size = float(dist.get_world_size())
+            for grad in gradients:
+                dist.all_reduce(grad.data, op=dist.reduce_op.SUM)
+                grad.data /= size
+
+        if gradients is not None:
+            params = list(params)
+            for p, g in zip(params, gradients):
+                if g is not None:
+                    p.add_(g, alpha=-lr)
+
+    def adapt(self, cloned_adaptation_net, train_loss):
+        fast_params = list(self.get_fast_params(cloned_adaptation_net))
+        self.update_params(
+            fast_params, train_loss, self.adaptation_lr, distributed=self.distributed
+        )
+
+    def clone_model(self, keep_as_reference=None):
+        """
+        Clones self.model by cloning some of the params and keeping those listed
+        specified `keep_as_reference` via reference.
+        """
+        model = clone_model(self.model.module, keep_as_reference=None)
+
+        if not self.distributed:
+            model = DataParallel(model)
+        else:
+            # Instead of using DistributedDataParallel, the grads will be reduced
+            # manually since we won't call loss.backward()
+            model
+
+        return model
+
+    def get_slow_params(self):
+        return self.model.module.slow_params
+
+    def get_fast_params(self, clone=None):
+        if clone is not None:
```

Comment:
Maybe just do
```
model = clone or self.model
...
```

Reply:
That just evaluates to True or False. I think you mean `model = clone if clone is not None else self.model`?

Reply:
Here's what we can do:
```
model = clone if clone is not None else self.module
if hasattr(model, "module"):
    return model.module.fast_params
else:
    return model.fast_params
```